### PR TITLE
Re-enable mmi mechas

### PR DESCRIPTION
**Pull request by Alberyk**
What the title says.

### CODE REVIEW

**Comment by BurgerLUA** [comment]
oh god

**Comment by Alberyk** [comment]
Laws are clearly controlled by a cyborg's component, that does not exist on mechas, so they are not lawed by default. Build it at your own risk.

File: code/game/mecha/mecha.dm [hunk 1]
```diff
@@ -1234,6 +1241,63 @@
 	else
 		return 0
 
+/obj/mecha/proc/mmi_move_inside(var/obj/item/device/mmi/mmi_as_oc as obj,mob/user as mob)
+	if(!src.operation_allowed(user))
+		usr << "<span class='warning'>Access denied</span>"
```
Comment (skull132) [inline code/game/mecha/mecha.dm]: `user` => `user`

File: code/game/mecha/mecha.dm [hunk 2]
```diff
@@ -1234,6 +1241,63 @@
...
+		user << "Beta-rhythm below acceptable level."
+		return 0
+	else if(occupant)
+		user << "Occupant detected."
+		return 0
+	else if(dna && dna!=mmi_as_oc.brainmob.dna.unique_enzymes)
+		user << "Stop it!"
+		return 0
+
+	visible_message("<span class='notice'>\The [usr] starts to insert \the [mmi_as_oc] into \the [src]</span>")
```
Comment (skull132) [inline code/game/mecha/mecha.dm]: `usr` => `user`

**Comment by Alberyk** [comment]
The trouble are not airlocks, or accessing them, it works just fine. The issue are using airlock buttons and controllers, I coded a way for mmi mechas to access buttons, but they can't just use airlock controllers, even if they can open the nanoui.

**Comment by LordFowl** [comment]
Probably a result of safety checks on the NanoUI side.

**Comment by Sindorman** [comment]
I remember that before when MMI was able to be in Mech, they were useless as the could only use airlocks that do not require special access.

**Comment by Alberyk** [comment]
This is not a thing anymore, you can add access to the mecha.

File: code/game/mecha/mecha.dm (hunk 1 above)
Comment (skull132) [reply code/game/mecha/mecha.dm]: `usr` => `user`

**Comment by Alberyk** [comment]
I will come back to this later.

**Pull request closed without merging**